packages/api-plugin-fulfillment/src/resolvers/Query/getFulfillmentMethods.js

```diff
@@ -0,0 +1,32 @@
+import getPaginatedResponseFromAggregate from "@reactioncommerce/api-utils/graphql/getPaginatedResponseFromAggregate.js";
+import wasFieldRequested from "@reactioncommerce/api-utils/graphql/wasFieldRequested.js";
+import { decodeShopOpaqueId, decodeFulfillmentGroupOpaqueId } from "../../xforms/id.js";
+
+/**
+ * @name Query/getFulfillmentMethods
+ * @method
+ * @memberof Fulfillment/Query
+ * @summary Query for a list of fulfillment methods
+ * @param {Object} _ - unused
+ * @param {Object} args - an object of all arguments that were sent by the client
+ * @param {String} args.shopId - Shop ID to get records for
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} info Info about the GraphQL request
+ * @returns {Promise<Object>} Fulfillment methods
+ */
+export default async function getFulfillmentMethods(_, args, context, info) {
+  const { shopId: opaqueShopId, fulfillmentTypeId: opaqueFulfillmentId, ...connectionArgs } = args;
+
+  const shopId = decodeShopOpaqueId(opaqueShopId);
```

Comment:
Should we be checking permissions in the resolver rather than in the method? Feels like we should? Can you check what we do elsewhere?

Reply:
I assume you're referring to `context.validatePermissions` and I see most of these calls happening in mutation/query methods

Reply:
@zenweasel Please let me know if this can be considered resolved